CEGameApp.API/Controllers/AuthController.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using CEGameApp.API.Data;
+using CEGameApp.API.Dtos;
+using CEGameApp.API.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+
+namespace CEGameApp.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AuthController : ControllerBase
+    {
+        private readonly IAuthRepository _repo;
+        private readonly IConfiguration _config;
+
+        public AuthController(IAuthRepository repo, IConfiguration config)
+        {
+            _repo = repo;
+            _config = config;
+        }
+
+        [HttpPost("register")]
+        public async Task<IActionResult> Register(UserForRegisterDto userForRegisterDto)
+        {
+            //validate request
+            userForRegisterDto.UserName = userForRegisterDto.UserName.ToLower();
+            
+            if (await _repo.UserExists(userForRegisterDto.UserName))
+                return BadRequest("Username already exists");
+            
+            var userToCreate = new Users
+            {
+                Name = userForRegisterDto.UserName
```

Comment:
so this is exactly why you don't want to inline mutate your UserName property.  now you are going to store it into your UserObject, and it will be different then what your user entered.  Also, you are violating the S in S.O.L.I.D.  this method is taking on too many responsibilities.  it now knows that a Users object exist.  also, make your User class lower class now plural.  a plural of User would be an IEnumerable<User>